plugins/webp-uploads/hooks.php

```diff
@@ -52,21 +52,30 @@
  * } An array with the updated structure for the metadata before is stored in the database.
  */
 function webp_uploads_create_sources_property( array $metadata, int $attachment_id ): array {
-	// This should take place only on the JPEG image.
+	$file = get_attached_file( $attachment_id, true );
+	// File does not exist.
+	if ( false === $file || ! file_exists( $file ) ) {
+		return $metadata;
+	}
+
+	/*
+	 * We need to get the MIME type ideally from the file, as WordPress Core may have already altered it.
+	 * The post MIME type is typically not updated during that process.
+	 */
+	$filetype = wp_check_filetype( $file );
+	if ( isset( $filetype['type'] ) ) {
+		$mime_type = $filetype['type'];
+	} else {
+		$mime_type = get_post_mime_type( $attachment_id );
+	}
```

Comment:
This can be simplified:

```suggestion
	$filetype  = wp_check_filetype( $file );
	$mime_type = $filetype['type'] ?? get_post_mime_type( $attachment_id );
	if ( ! is_string( $mime_type ) ) {
		return $metadata;
	}
```

Also applying the suggestion from @felixarntz in https://github.com/WordPress/performance/pull/1635/files#r1832951586